plugins/catalog-react/src/alpha.tsx

```diff
@@ -37,21 +36,57 @@ export { useEntityPermission } from './hooks/useEntityPermission';
 export const entityContentTitleExtensionDataRef =
   createExtensionDataRef<string>('plugin.catalog.entity.content.title');
 
+/** @alpha */
+export const entityFilterExtensionDataRef = createExtensionDataRef<
+  (ctx: { entity: Entity }) => boolean
+>('plugin.catalog.entity.filter');
+
+function applyFilter(a?: string, b?: string): boolean {
+  if (!a) {
+    return true;
+  }
+  return a.toLocaleLowerCase('en-US') === b?.toLocaleLowerCase('en-US');
+}
+
+// TODO: Only two hardcoded isKind and isType filters are available for now
+//       This is just an initial config filter implementation and needs to be revisited
+function buildFilter(
+  config: { filter?: { isKind?: string; isType?: string }[] },
+  filterFunc?: (ctx: { entity: Entity }) => boolean,
+) {
+  return (ctx: { entity: Entity }) => {
+    const configuredFilterMatch = config.filter?.some(filter => {
+      const kindMatch = applyFilter(filter.isKind, ctx.entity.kind);
+      const typeMatch = applyFilter(
+        filter.isType,
+        ctx.entity.spec?.type?.toString(),
+      );
+      return kindMatch && typeMatch;
```

Comment:
Do you this comparison could be hard to maintain depending on the number of filters we would like to apply? I was wondering if having a filter chain processor is worth it...

Reply:
Yep for sure, this is intended to be refactored into something sensible